### ISSUE TITLE
Removal Candidate

### ISSUE THREAD

**Issue by Adurol** [KaOS-Community-Packages/cairo-dock-plugins#1]
This Package will be removed in from KCP.

This is practically a blind copy of the [AUR PKGBUILD](https://www.archlinux.org/packages/community/x86_64/cairo-dock-plug-ins/), the same issue as [cairo-dock](https://github.com/KaOS-Community-Packages/cairo-dock/issues/1). But here I even not got it to build. I get the following failure:
```
CMake Error at /usr/share/cmake-3.7/Modules/FindPkgConfig.cmake:415 (message):
  A required package was not found
Call Stack (most recent call first):
  /usr/share/cmake-3.7/Modules/FindPkgConfig.cmake:588 (_pkg_check_modules_internal)
  CMakeLists.txt:52 (pkg_check_modules)


-- Configuring incomplete, errors occurred!
See also "/home/twohalf/Development/build/KCP/test/src/cairo-dock-plug-ins-3.4.1/build/CMakeFiles/CMakeOutput.log".
==> ERROR: A failure occurred in build().
    Aborting...
```
Also please check the naming of the package. For KCP it is named cairo-dock-__plugins__ but  `cairo-dock-plug-ins` in the PKGBUILD (extra "-"). Booth names must be identical for KCP.

Please read the [KCP Guide](https://kaosx.us/docs/kcp/#submitting-a-package) first. Only submit packages that were adjusted for KaOS or you created yourself for KaOS. Blindly copying any existing PKGBUILD and submitting it to KCP is not the reason it was created for. If you can find working PKGBUILDs elsewhere, use them.

**Comment by demmm** [comment]
Let's not remove, but many corrections needed.
Starting with consistent naming, moving the opdepends just about all to depends.
And checking all actual needed depends.

**Comment by kmiii** [comment]
I have changed the PKGBUILD I have adapted it just with the required dependencies of the official web http://glx-dock.org/ww_page.php?p=By%20compiling&lang=en, I have also changed the build instructions.

Can you check if everything is ok?

**Comment by Adurol** [comment]
I got the dock build (but didn't try them, so no idea if they work as intended), but not the plugins.

* Add to the readme.md that plugins need to be installed with as depends enforced with `kcp`
* Set `cairo-dock` as depends (without it, plugins can't be build!)
* Why are the depends of plugins almost the same as for the dock? If plugins depends on dock, no need to set it again.
* What about the optdepends?
* Why is `bzr` in the makedepends? (same for dock)
* I know `make -j` is in the official docs, but is it really needed? (same for dock)
* Did you check with `ldd`?

`cairo-dock-plugins` has a bunch of optional dependencies. And mostly specific features are only available if you have the dependencies installed. So it would be good to set a sane default of optional dependencies as "hard" dependencies to get a good user experience (you will have to think about that). The other thing with this is, how to enable specific features? Can you enforce them (or deactivate unnecessary f.e. unity-specific features) per cmake? Or is the makefile so good written that it will do the things "magically" for you?

**Comment by kmiii** [comment]
> Add to the readme.md that plugins need to be installed with as depends enforced for kcp

ok, done
> Set cairo-dock as depends (without it, plugins can't be build!)

ok, done
>Why are the depends of plugins almost the same as for the dock? If plugins depends on dock, no need to set it again.

ok, done
> What about the optdepends?

I have checked all optdepends and are for gnome desktop

> Why is bzr in the makedepends? (same for dock)

ok, done
>I know make -j is in the official docs, but is it really needed?

ok, done
>Did you check with ldd?

yes I checked
> The other thing with this is, how to enable specific features? Can you enforce them (or deactivate unnecessary f.e. unity-specific features) per cmake? Or is the makefile so good written that it will do the things "magically" for you?

Specific features are selected in the dock configuration.
The makefile only compiles the features found in the system, those that do not find them ignore

I think that now everything is correct, the program compiles and works correctly in kaos and installs perfectly from kcp

**Comment by Adurol** [comment]
Fist of all: **Please write better commit messages!** As they are now, these are total meaningless. I have no idea on a quick glace what you have done. I have to look into the diff every time...
Also git supports pretty features like squashing commits and other nifty things to make commits more presentable to the "outside" (where you push to).

---

> The other thing with this is, how to enable specific features? Can you enforce them (or deactivate unnecessary f.e. unity-specific features) per cmake? Or is the makefile so good written that it will do the things "magically" for you?

>> Specific features are selected in the dock configuration.
>> The makefile only compiles the features found in the system, those that do not find them ignore
>>
>> I think that now everything is correct, the program compiles and works correctly in kaos and installs perfectly from kcp

OK. I don't think you got what I meant.

Yes, you can enable features in Cairo-Dock. And if you have `cairo-dock-plugins` installed, even the features of the Plugins. BUT, if you do it the way you suggested ("The makefile only compiles the features found in the system, those that do not find them ignore") than every user will find other features he can use. Some have maybe a package installed the other has not. That is not good style.
The solution to this is to set some packages as hard depends. This way, the user will have to install specific packages and will get a minimum feature set. *What* packages are to be set is up to you. On the website, you find the list of dependencies `cairo-dock-plugins` can use. I don't use Cairo-Dock so I can't say anything about it.

The second thing about that is, as I said, *how* to enable or disable specific features (the disable part is more important). Yes, you can just let the makefile do the decision. But that is not always the best way. You will have to look into the makefile (or CMakeLists.txt) to deactivate useless features that won't be of any need on KaOS but are activated by default (that's why the example about unity-specific features). Or enforce to enable features that are deactivated by default but are useful in your opinion.

Just copy-paste a PKGBUILD without understanding how the process of building a software works is a pretty steep start. I don't say you have to know the ins and outs, but you should be able to read a makefile, know how to interpret instructions of the official website / readme and (in the worst case) know how to patch the software to get it running. There is plenty of tutorials for writing a PKGBUILD out there.

**Comment by kmiii** [comment]
Please check it now.

**Comment by Adurol** [comment]
No, how you did it in @3cc2ad7979941909df68079b82f23bd116b5cb46 is not how it is supposed to be.

You're PGKBUILD will:
* not have optdepends
* most likely not have makedepends
* most likely have depends

If you don't know what each of thees variables do, please look them up. `man` is your friend.

Please read again what demm and I have written (and also discussed on IRC) and think about it again! You don't seem to understand what I'm trying to say.

If you don't understand how PKGBUILDs work, best drop by in the [forum](http://forum.kaosx.us/) and start a thread asking on how a PKGBUILD should be made. GitHub Issues are more for solving and discussing immanent problems, not about teaching how it should be done. I (and I think demm too) won't give you a ready-made PKGBUILD. You will have to learn how to create those (and something like this is better done in the forum than here.)

**Comment by kmiii** [comment]
Ok, thank you very much for your information, I take note of everything and I will read and learn about the PKGBUILD, it will take me a while. If you think necessary, please remove cairo-dock, cairo-dock-plugins and mame of kcp, until I know how to created a correct PKGBUILD.

**Comment by Adurol** [comment]
No, with @cf26b333e9d2882827df00638c74f8de5099a671 you didn't what I was telling you again... You're clearly lacking knowledge how packaging works, so here my suggestion:

You're packages packages will be removed from KCP. You open a thread for your application you want to get into KCP on the [forum](https://forum.kaosx.us/) (you can login with GitHub). Learn in the forum, how packaging for KaOS works. And when you got the knowledge how to write and maintain a package, submit it to KCP again.
